index.js

```diff
@@ -38,17 +37,16 @@ function _make_generateLog(metrics, func, start_time, config, context) {
   var pre_stat_promise = system.readstat('self')
 
   return function generateLog(err, callback) {
-    Promise.join(
+    Promise.all([
         pre_stat_promise,
         system.readstat('self'),
         system.readstatus('self'),
         system.readbootid()
-      ).spread((
-        pre_proc_self_stat,
-        proc_self_stat,
-        proc_self_status,
-        boot_id
-      ) => {
+      ]).then((results) => {
+        const [pre_proc_self_stat,
```

Comment:
I'm so ready to have nice things yesterday (or pursuing the babelization of this lib, either or). Thanks for the heads up